curriculum/challenges/english/25-front-end-development/workshop-merge-sort/6569c2cbf6c993ea8cd85682.md

```diff
@@ -0,0 +1,121 @@
+
+With that, the merge sort algorithm is complete.
+
+# --hints--
+
+You should have `print('Sorted array: ' + str(numbers))` in your code. Remember to add the space after the colon.
+
+```js
+({
+    test: () => assert(runPython(`
+      _Node(_code).find_ifs()[0].has_call("print('Sorted array: ' + str(numbers))")`))
+})
+```
+
+# --seed--
+
+## --seed-contents--
+
+```py
+def merge_sort(array):
+    if len(array) <= 1:
+        return
+    
+    middle_point = len(array) // 2
+    left_part = array[:middle_point]
+    right_part = array[middle_point:]
+
+    merge_sort(left_part)
+    merge_sort(right_part)
+
+    left_array_index = 0
+    right_array_index = 0
+    sorted_index = 0
+
+    while left_array_index < len(left_part) and right_array_index < len(right_part):
+        if left_part[left_array_index] < right_part[right_array_index]:
+            array[sorted_index] = left_part[left_array_index]
+            left_array_index += 1
+        else:
+            array[sorted_index] = right_part[right_array_index]
+            right_array_index += 1
+        sorted_index += 1
+
+    while left_array_index < len(left_part):
+        array[sorted_index] = left_part[left_array_index]
+        left_array_index += 1
+        sorted_index += 1
+    
+    while right_array_index < len(right_part):
+        array[sorted_index] = right_part[right_array_index]
+        right_array_index += 1
+        sorted_index += 1
+
+
+--fcc-editable-region--
+if __name__ == '__main__':
+    numbers = [4, 10, 6, 14, 2, 1, 8, 5]
+    print('Unsorted array: ')
+    print(numbers)
+    merge_sort(numbers)
+
+--fcc-editable-region--
+```
+
+# --solutions--
+
+```py
+def merge_sort(array):
+    if len(array) <= 1:
+        return
+    
+    middle_point = len(array) // 2
+    left_part = array[:middle_point]
+    right_part = array[middle_point:]
+
+    merge_sort(left_part)
+    merge_sort(right_part)
+
+    left_array_index = 0
+    right_array_index = 0
+    sorted_index = 0
+
+    while left_array_index < len(left_part) and right_array_index < len(right_part):
+        if left_part[left_array_index] < right_part[right_array_index]:
+            array[sorted_index] = left_part[left_array_index]
+            left_array_index += 1
+        else:
+            array[sorted_index] = right_part[right_array_index]
+            right_array_index += 1
+        sorted_index += 1
+
+    while left_array_index < len(left_part):
+        array[sorted_index] = left_part[left_array_index]
+        left_array_index += 1
+        sorted_index += 1
+
+    while right_array_index < len(right_part):
+        array[sorted_index] = right_part[right_array_index]
+        right_array_index += 1
+        sorted_index += 1
+
+
+if __name__ == '__main__':
+    numbers = [4, 10, 6, 14, 2, 1, 8, 5]
+    print('Unsorted array: ')
+    print(numbers)
+    merge_sort(numbers)
+    print('Sorted array: ' + str(numbers))
```

Comment:
Not a big deal but it's very annoying to see the same thing done in two different ways here.
It would make sense to have either:

```py
print('Unsorted array: ')
print(numbers)
merge_sort(numbers)
print('Sorted array: ')
print(numbers)
```
or

```py
print('Unsorted array: ' + str(numbers)) 
merge_sort(numbers)
print('Sorted array: ' + str(numbers))
```
or even

```py
print(f'Unsorted array: {numbers}') 
merge_sort(numbers)
print(f'Sorted array: {numbers}')
```

Reply:
I chose the first one and updated the solution and hint.